graylog2-server/src/main/java/org/graylog2/alerts/AbstractAlertCondition.java

```diff
@@ -155,6 +160,32 @@ public boolean shouldRepeatNotifications() {
         return repeatNotifications;
     }
 
+    /**
+     * Combines the given stream ID and query string into a single filter string.
+     *
+     * @param streamId the stream ID
+     * @param query    the query string (might be null or empty)
+     * @return the combined filter string
+     */
+    protected String buildQueryFilter(String streamId, String query) {
+        checkArgument(streamId != null, "streamId parameter cannot be null");
+
+        final String trimmedStreamId = streamId.trim();
+
+        checkArgument(!trimmedStreamId.isEmpty(), "streamId parameter cannot be empty");
+
+        final StringBuilder builder = new StringBuilder().append("streams:").append(trimmedStreamId);
+
+        if (query != null) {
+            final String trimmedQuery = query.trim();
+            if (!trimmedQuery.isEmpty() && !"*".equals(trimmedQuery)) {
+                builder.append(" ").append(trimmedQuery);
```

Comment:
Shouldn't we join both filters with an `AND` operator? Otherwise this does not further filter messages in the stream, as the user would expect.